src/main/java/com/wcc/platform/service/ProgrammeService.java

```diff
@@ -0,0 +1,41 @@
+package com.wcc.platform.service;
+
+import static com.wcc.platform.domain.cms.ApiResourcesFile.PROG_BOOK_CLUB;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.wcc.platform.domain.cms.attributes.ProgramType;
+import com.wcc.platform.domain.cms.pages.programme.ProgrammePage;
+import com.wcc.platform.domain.exceptions.PlatformInternalException;
+import com.wcc.platform.utils.FileUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/** Programme Service. */
+@Service
+public class ProgrammeService {
+
+  private final ObjectMapper objectMapper;
+
+  @Autowired
+  public ProgrammeService(final ObjectMapper objectMapper) {
+    this.objectMapper = objectMapper;
+  }
+
+  /**
+   * API to fetch details about different type of programmes.
+   *
+   * @return Programme Page json response
+   */
+  public ProgrammePage getProgramme(final ProgramType programType) {
+    try {
+      String data = null;
+      if (ProgramType.BOOK_CLUB.equals(programType)) {
```

Comment:
you don't need the programType as parameter here, we need validation in the controller level (which can be added later) 

but in the service we know that is for bookClub

Reply:
@dricazenck - I was reading through the best practices, I suppose the programme type validation should happen at the service layer rather than at the controller level - https://www.baeldung.com/spring-service-layer-validation
What do you think ?

Reply:
We can discuss and decide if this needs to be changed.